engine/dlib/src/test/test_log.cpp

```diff
@@ -258,6 +259,51 @@ TEST(dmLog, TestCapture)
     ASSERT_STREQ(ExpectedOutput, g_LogListenerOutput.Begin());
 }
 
+
+static void LogThreadWithLogCalls(void* arg)
+{
+    int* run = (int*)arg;
+    int count = 0;
+    while (*run)
+    {
+        dmLogError("a warning %d", count);
+        count++;
+    }
+}
+
+int g_LogThreadWithLogCallsCount = 0;
+static void LogThreadWithLogCallsListener(LogSeverity severity, const char* domain, const char* formatted_string)
+{
+    g_LogThreadWithLogCallsCount++;
+}
+
+TEST(dmLog, TestLogThreadWithLogCalls)
+{
+    int run = 1;
+    dLib::SetDebugMode(false); // avoid spam in the unit tests
+
+    dmThread::Thread log_thread = dmThread::New(LogThreadWithLogCalls, 0x80000, (void*)&run, "test");
+
+    dmLog::LogParams params;
+    dmLog::LogInitialize(&params);
+
+
+    g_LogThreadWithLogCallsCount = 0;
+    dmLogRegisterListener(LogThreadWithLogCallsListener);
+
+    while (g_LogThreadWithLogCallsCount < 40)
+        dmTime::Sleep(1000);
+
+    dmLog::LogFinalize();
+
+    dmTime::Sleep(1000); // make sure we write some more logs
```

Comment:
Here we make sure we log a lot of info before and after the log system is initialized/finalized.

Without the fixes, it would crash on using the server pointer after deletion.